lua/core/menu/system.lua

```diff
@@ -21,21 +27,31 @@ end
 
 m.redraw = function()
   screen.clear()
-
-  for i=1,#m.list do
-    screen.move(0,10*i)
-    if(i==m.pos) then
-      screen.level(15)
-    else
-      screen.level(4)
+  for i=1,6 do
+    if (i > 3 - m.pos) and (i < #m.list - m.pos + 4) then
+      screen.move(0,10*i)
+      local line = m.list[i+m.pos-3]
+      if(i==3) then
+        screen.level(15)
+      else
+        screen.level(4)
+      end
+      screen.text(line)
     end
-    screen.text(m.list[i])
   end
-
   screen.update()
 end
 
 m.init = norns.none
 m.deinit = norns.none
 
+m.passdone = function(txt)
+  if txt ~= nil then
+    print("password change to " .. txt)
+    os.execute("echo 'we:"..txt.."' | sudo chpasswd")
```

Comment:
suggest checking that return value of `os.execute(...)` and confirming that the command did succeed then simply printing out whether the password change succeeded or failed. print the actual password in cleartext would end up writing it to the system logs and over the network (unencrypted) if maiden is open.

Reply:
lol i guess forgotten prints are a common security hole